pallets/xcm-asset-manager/src/lib.rs

```diff
@@ -145,6 +147,8 @@ pub mod pallet {
 		type Currency: ReservableCurrency<Self::AccountId>;
 
 		type WeightInfo: WeightInfo;
+
+		type NativeTokenForeignAssetType: Get<ForeignAssetTypeOf<Self>>;
```

Comment:
It seems this type is never used in the business logic except in the `genesis build`?

Reply:
yes. For genesis build only.

Reply:
That doesn't make much sense to me 😂  I would expect something like "you can't register asset type that is equal to NativeTokenForeignAssetType" at least..

Reply:
What you describe is partial true, we actually can not serialized multilocation, so genesis build can not use NativeTokenForeignAssetType directly.

Reply:
I mean if it's only used in genesis build, what's the point of having it in the future logic?